slatedb/src/compactor_executor.rs

```diff
@@ -221,14 +221,19 @@ impl TokioCompactionExecutorInner {
     }
 
     fn stop(&self) {
-        let mut tasks = self.tasks.lock();
-
-        for task in tasks.values() {
-            task.task.abort();
-        }
+        // Drain all tasks and abort them, then release tasks lock so
+        // the cleanup function in spawn_bg_task (above) can take the
+        // lock and remove the task from the map.
+        let task_handles = {
```

Comment:
ah good catch. I guess this was introduced when I added the background cleanup stuff.